### PR TITLE
#9 added filter implementation basically

### PR DESCRIPTION
**Pull request by hkulekci**
I did not execute my test cases because I dont know how to configure my GOPATH correctly. (dam it!) But I tried to simulate with a main function on go playground that is really cool tool and it worked. And this is my first golang code : https://play.golang.org/p/akic0tGDLg 

There is some unimplemented point in this pr. But I separate them and we can continue without. I'll try to implement this parts following days.

### CODE REVIEW

File: token_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestAsciiFoldingTokenFilter(t *testing.T) {
+	assert := assert.New(t)
+}
```
Comment (hevalberknevruz) [inline token_filter_test.go]: Unnecessary test method.

File: token_filter.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func (input *TokenizerOutput) Filter() *TokenFilterOutput {
+	filteredTokens := []string{}
+
+	for _, r := range input.Tokens {
+		//find more generic solution for here.
+		filteredToken := AsciiFoldingTokenFilter(r)
+		filteredToken = LowercaseTokenFilter(filteredToken)
+		filteredToken = StopWordTokenFilter(filteredToken)
+
```
Comment (hevalberknevruz) [inline token_filter.go]: To me, Each filter must be independent and filters must be selected in analyzer. (Analyzers are composed of a single Tokenizer and zero or more TokenFilters. The tokenizer may be preceded by one or more CharFilters.)

So, each filter we separately must create.

Comment (hevalberknevruz) [inline token_filter.go]: But initially If you want to be in this way. Maybe, later we can arrange

Comment (hevalberknevruz) [inline token_filter.go]: @hkulekci Continuous Integration tool sends notification that If it be fail of you tests

File: token_filter_test.go (hunk 1 above)
Comment (manorie) [inline token_filter_test.go]: assert.New kaldirsan, ilk draft olarak yeter bence. Thumbs up caktim ben.

**Pull request merged**